Marlin/src/lcd/dogm/u8g_dev_uc1701_mini12864_HAL.cpp

```diff
@@ -119,7 +120,21 @@ static const uint8_t u8g_dev_uc1701_mini12864_HAL_init_seq[] PROGMEM = {
 static const uint8_t u8g_dev_uc1701_mini12864_HAL_data_start[] PROGMEM = {
   U8G_ESC_ADR(0),             /* instruction mode */
   U8G_ESC_CS(1),              /* enable chip */
-  UC1701_COLUMN_ADR(0),       /* address 0 */
+  #if ENABLED(MKS_MINI_12864)
+    UC1701_START_LINE(0),     /* set display start line to 0 */
+    UC1701_ADC_REVERSE(0),    /* ADC set to reverse */
+    UC1701_OUT_MODE(1),       /* common output mode */
+    UC1701_INVERTED(0),       /* display normal, bit val 0: LCD pixel off. */
+    UC1701_BIAS_MODE(0),      /* LCD bias 1/9 */
+    UC1701_POWER_CONTROL(0x7), /* all power control circuits on */
+    UC1701_BOOST_RATIO(0x0),  /* set booster ratio to 4x */
+    UC1701_V5_RATIO(3),       /* set V0 voltage resistor ratio to large */
+    UC1701_INDICATOR(0),      /* indicator disable */
+    UC1701_ON(1),             /* display on */
+    UC1701_COLUMN_HI(0),      /* set upper 4 bits of the col adr to 0 */
+  #else
```

Comment:
Data packets to the display should be as small as possible. Adding this much header information to every data packet is surely not a requirement (what does the datasheet say?). Is it possible that on your particular LCD there is static or other electrical noise spuriously corrupting the internal settings of the display?

See if you can remove some (most) of these extra commands to narrow down the minimum ones required to produce a stable display. Consider whether the power or other settings are being initialized incorrectly…. Refer to the data-sheet and ensure a good solid LCD connection.

Reply:
This additional bytes acts as small delay that compensate the hi switching frecuency at CS line

Reply:
@thinkyhead 
This issue was observed by different people and on 3 different boards - STM32F407VE, Re-ARM and SKR 1.x
All boards have 100+ MHz MCU. MKS Mini 12864 is known to be stable with 72MHz STM32F103.
And these 'extra' data are send in every 8th packet - at the beginning of screen refresh cycle.

@DrDitto
I suggest to add F_CPU check to ensure that these 'extra' data is only send for fast MCUs that cause stability problems for MKS Mini 12864.

Reply:
Did you try adjusting the SPI delay options before adding all this extra code to create an artificial delay?
```cpp
#define ST7920_DELAY_1 DELAY_NS(125)
#define ST7920_DELAY_2 DELAY_NS(125)
#define ST7920_DELAY_3 DELAY_NS(125)
```

Reply:
point is mks mini mini 12864 uses uc1701 not st920